File: 14/go/main.go
package main


import (
    "fmt"
)


func longestCommonPrefix(strs []string) string {
    res := ""
    length := 200
    for _, v := range strs {
        length = min(length, len(v))
    }
    loop: for i := range length {
        curr := strs[0][i]
        for _, s := range strs[1:] {
            if s[i] != curr {
                break loop
            }
        }
        res += string(curr)
    }
    return res
}


func main() {
    strs := []string{"flower","flow","flight"}
    res := longestCommonPrefix(strs)
    fmt.Println(res)
}

File: 14/rust/src/main.rs
struct Solution {}


impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        let mut shortest: usize = 200;
        for s in &strs {
            shortest = if shortest < String::len(&s) {shortest} else {String::len(&s)};
        }
        let mut res: Vec<u8> = vec!();
        'outer: for i in 0..shortest {
            let curr = strs[0].as_bytes()[i];
            for s in 1..strs.len() {
                if strs[s].as_bytes()[i] != curr {
                    break 'outer;
                }
            }
            res.push(curr);
        }
        String::from_utf8(res).unwrap()
    }
}


fn main() {
    let strs = ["flower","flow","flight"];
    let strs: Vec<String> = strs.into_iter().map(|s| {s.to_string()}).collect();
    let res: String = Solution::longest_common_prefix(strs);
    println!("{res}");
}
